### ISSUE TITLE
high cpu

### ISSUE THREAD

**Issue by atian25** [Microsoft/vscode#21442]
- VSCode Version: Code 1.9.1  (f9d0c687ff2ea7aabd85fb9a43129117c0ecf519, 2017-02-08T23:31:51.320Z)
- OS Version: Darwin x64 16.3.0
- Extensions:

|Extension|Author|Version|
|---|---|---|
|vscode-mocha|Compulim|0.1.1|
|output-colorizer|IBM|0.0.11|
|vscode-file-peek|abierbaum|1.0.1|
|Bookmarks|alefragnani|0.11.0|
|project-manager|alefragnani|0.13.2|
|copy-syntax|atian25|0.5.0|
|vscode-custom-css|be5invis|2.2.0|
|npm-intellisense|christian-kohler|1.3.0|
|vscode-eslint|dbaeumer|1.2.6|
|vscode-npm-source|dkundel|1.0.0|
|code-runner|formulahendry|0.6.12|
|task-master|ianhoney|0.1.39|
|path-autocomplete|ionutvmi|1.4.1|
|plantuml|jebbs|1.2.3|
|docthis|joelday|0.4.0|
|md-navigate|jrieken|0.0.1|
|vue|liuji-jim|0.1.5|
|debugger-for-chrome|msjsdiag|2.5.4|
|node-modules-resolve|naumovs|1.0.2|
|peep|nwallace|0.0.5|
|vscode-json-transform|octref|0.1.2|
|unittest-navigate|roblourens|0.0.2|
|preview-vscode|searKing|1.1.1|
|code-navigation|vikas|0.2.0|
|JavaScriptSnippets|xabikos|1.4.0|

---

```json
// Place your settings in this file to overwrite the default settings
{
  "extensions.autoUpdate": true,
  "editor.tabSize": 2,
  "editor.renderIndentGuides": true,
  "editor.fontFamily": "Menlo, Monaco, 'Courier New', monospace",
  "editor.formatOnPaste": true,

  "files.trimTrailingWhitespace": true,
  "files.autoSave": "onFocusChange",
  "files.hotExit": "onExitAndWindowClose",
  "files.associations": {
    "untitled-*": "javascript"
  },
  "files.exclude":{
    "**/node_modules/.*@*": true,
    "**/node_modules/babel-*": true,
    "**/node_modules/power-assert-*": true,
    "**/.idea": true,
    ".vscode-test": true
  },

  "search.exclude": {
    "**/logs": true,
    "**/run": true,
    "**/coverage/**": true,
    // "**/node_modules": true,
    // "**/node_modules/.*@*": true,
    // "**/node_modules/babel-*": true,
    // "**/node_modules/power-assert-*": true,
    // "**/node_modules/(!egg*)": true,
    // "**/node_modules/**/node_modules/**": true
  },

  "javascript.format.insertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets": true,

  "terminal.external.osxExec": "iTerm.app",

  "mocha.options": {
    "timeout": "30000"
  },
  "mocha.files.glob": "test/**/*.test.js",

  "html.suggest.angular1": false,
  "html.suggest.ionic": false,

  "code-runner.defaultLanguage": "javascript",
  "code-runner.clearPreviousOutput": true,
  "code-runner.runInTerminal": true,

  "vscode_custom_css.imports": ["~/.vscode/custom.css"],

  "copy-syntax.showMessage": false
}
```

Steps to Reproduce:

1. do nth but just open some project
2. had try `--disable-extensions` but still the same
3. seem to be index/search issue, when remove `node_modules` it work fine. but had disable `node_modules`
4. any tool to detect this like `Chrome Task Manager`?

![image](https://cloud.githubusercontent.com/assets/227713/23338909/7786dc9e-fc51-11e6-8ff7-78f199bf061d.png)

**Comment by bpasero** [comment]
To find out more about this issue, we need to get at the process that is causing the high load. VS Code creates multiple processes, each with different tasks. Depending on your OS you can get at the full command line of the process and paste it here. On Mac, the **Activity Monitor** allows to find the `PID` of the process and from a terminal `ps aux | grep <pid>` reveals its arguments. 

![image](https://cloud.githubusercontent.com/assets/900690/18907063/65806550-856a-11e6-8b2e-83da9111445d.png)

On Windows a tool like **[Process Explorer](https://technet.microsoft.com/en-us/sysinternals/processexplorer.aspx)** helps:

![image](https://cloud.githubusercontent.com/assets/900690/18907037/476f0aa8-856a-11e6-8ac3-4c1f3e803f32.png)

Once you have the information please update this issue with it. Thanks!

**Comment by rimtd** [comment]
see the pic blow
![qq 20170226212918](https://cloud.githubusercontent.com/assets/1570512/23340093/7ba56c18-fc6b-11e6-9e1a-4da99b8f0281.png)

**Comment by bpasero** [comment]
Unless @atian25 and @rimtd are the same people I would like to see @atian25 result. Any process could be high CPU, it is not the same issue for everyone.

**Comment by atian25** [comment]
when I remove comment from settings, don't get cpu 100% tonight, I will try again tomorrow.

```
   // "**/node_modules": true,
    // "**/node_modules/.*@*": true,
    // "**/node_modules/babel-*": true,
    // "**/node_modules/power-assert-*": true,
    // "**/node_modules/(!egg*)": true,
    // "**/node_modules/**/node_modules/**": true
```

PS:  when using comment at settings.json, will it ignore or invalid(then use last setting which is not comment?)

**Comment by bpasero** [comment]
@atian25 ok that sounds like your issue is related to running a search.

**Comment by atian25** [comment]
after restore search rule:
```
  "search.exclude": {
    "**/logs": true,
    "**/run": true,
    "**/coverage/**": true,
    "**/node_modules": false,
    "**/node_modules/.*@*": true,
    "**/node_modules/babel-*": true,
    "**/node_modules/power-assert-*": true,
    "**/node_modules/(!egg*)": true,
    "**/node_modules/**/test/**": true,
    "**/node_modules/**/node_modules/**": true
  },
```

then I got 
![image](https://cloud.githubusercontent.com/assets/227713/23345532/b92a8870-fcca-11e6-9ebf-40b7c5dc9734.png)

```
➜  egg-init git:(npm-info-404) ps aux | grep 14446
tz               14446  99.0  0.8  3288316 134756   ??  R     6:33PM  15:33.13 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/node_modules/typescript/lib/typingsInstaller.js --globalTypingsCacheLocation /Users/tz/Library/Caches/typescript --enableTelemetry
```

**Comment by atian25** [comment]
`typingsInstaller.js --globalTypingsCacheLocation /Users/tz/Library/Caches/typescript` seems to be `typing auto install problem`

**Comment by bpasero** [comment]
@atian25 thanks, assigning to @mjbvz and @waderyan since this seems to be a TS issue.

**Comment by atian25** [comment]
thanks, any workaround for now ?

**Comment by mjbvz** [comment]
@atian25 Please try collecting a trace of the TypeScript Server when this occurs. Here's information on to do this using the TSS_LOG environment variable: https://github.com/Microsoft/TypeScript/wiki/Standalone-Server-%28tsserver%29#logging

From a command line, run:

Window:
```
set TSS_LOG=-level verbose -file c:\tmp\tsserver.log
code 
```

OSX:
```
$ export TSS_LOG="-level verbose -file /Users/${username-`whoami`}/tsserver.log"
$ code 
```

Reproduce the problem in VSCode and then share the `tsserver.log` file

Let me know if you have any questions about this.



As a work around, you may also want to try upgrading to VSCode 1.10 (the current insiders builds). This  release includes a new version of TypeScript that fixes some issues like this. If you are still blocked by this,  try to disable automatic typings acquisition using the `"typescript.disableAutomaticTypeAcquisition": true` setting. This will impact intellisense for packages, but hopefully will keep you from hitting the perf bug

**Comment by atian25** [comment]
```bash
Info 0    Binding...
Info 1    request: {"seq":0,"type":"request","command":"configure","arguments":{"hostInfo":"vscode"}}
Info 2    Host information vscode
Info 3    response: {"seq":0,"type":"response","command":"configure","request_seq":0,"success":true}
Perf 4    0::configure: async elapsed time (in milliseconds) 2.0564
Info 5    request: {"seq":1,"type":"request","command":"compilerOptionsForInferredProjects","arguments":{"options":{"module":"CommonJS","target":"ES6","allowSyntheticDefaultImports":true,"allowNonTsExtensions":true,"allowJs":true,"jsx":"Preserve"}}}
Perf 6    1::compilerOptionsForInferredProjects: elapsed time (in milliseconds) 0.8545
Info 7    response: {"seq":0,"type":"response","command":"compilerOptionsForInferredProjects","request_seq":1,"success":true,"body":true}
Info 8    request: {"seq":2,"type":"request","command":"getSupportedCodeFixes","arguments":null}
Perf 9    2::getSupportedCodeFixes: elapsed time (in milliseconds) 0.6030
Info 10   response: {"seq":0,"type":"response","command":"getSupportedCodeFixes","request_seq":2,"success":true,"body":["2377","17009"]}
Info 11   request: {"seq":3,"type":"request","command":"getSupportedCodeFixes","arguments":null}
Perf 12   3::getSupportedCodeFixes: elapsed time (in milliseconds) 0.1215
Info 13   response: {"seq":0,"type":"response","command":"getSupportedCodeFixes","request_seq":3,"success":true,"body":["2377","17009"]}
Info 14   request: {"seq":4,"type":"request","command":"getSupportedCodeFixes","arguments":null}
Perf 15   4::getSupportedCodeFixes: elapsed time (in milliseconds) 0.0791
Info 16   response: {"seq":0,"type":"response","command":"getSupportedCodeFixes","request_seq":4,"success":true,"body":["2377","17009"]}
Info 17   request: {"seq":5,"type":"request","command":"getSupportedCodeFixes","arguments":null}
Perf 18   5::getSupportedCodeFixes: elapsed time (in milliseconds) 0.0555
Info 19   response: {"seq":0,"type":"response","command":"getSupportedCodeFixes","request_seq":5,"success":true,"body":["2377","17009"]}
Info 20   request: {"seq":6,"type":"request","command":"open","arguments":{"file":"/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js","fileContent":"'use strict';\n\nconst path = require('path');\nconst Program = require('../../../..').Program;\n\nclass MyProgram extends Program {\n  constructor() {\n    super();\n    this.version = require('../package.json').version;\n\n    this.addCommand('test', path.join(__dirname, 'test_command.js'));\n  }\n}\n\nmodule.exports = MyProgram;\n"}}
Info 21   Search path: /Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib
Info 22   No config files found.
Info 23   Add watcher for: /Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib
Info 24   Add watcher for: /Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin
Info 25   Add watcher for: /Users/tz/Workspaces/eggjs/common-bin/test/fixtures
Info 26   Add watcher for: /Users/tz/Workspaces/eggjs/common-bin/test
Info 27   Add watcher for: /Users/tz/Workspaces/eggjs/common-bin
Info 28   Add watcher for: /Users/tz/Workspaces/eggjs
Info 29   Add watcher for: /Users/tz/Workspaces
Info 30   Add watcher for: /Users/tz
Info 31   Add watcher for: /Users
Info 32   Scheduling throttled operation: {"projectName":"/dev/null/inferredProject1*","fileNames":["/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/node_modules/typescript/lib/lib.es6.d.ts","/Users/tz/Workspaces/eggjs/common-bin/lib/run.js","/Users/tz/Workspaces/eggjs/common-bin/lib/program.js","/Users/tz/Workspaces/eggjs/common-bin/lib/helper.js","/Users/tz/Workspaces/eggjs/common-bin/lib/command.js","/Users/tz/Workspaces/eggjs/common-bin/index.js","/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js"],"compilerOptions":{"module":1,"target":2,"allowSyntheticDefaultImports":true,"allowNonTsExtensions":true,"allowJs":true,"jsx":1,"noEmitForJsFiles":true},"typeAcquisition":{"enable":true,"include":[],"exclude":[]},"unresolvedImports":["assert","child_process","child_process","crypto","events","fs","http","net","path","path","readline","stream","tls"],"projectRootPath":"","kind":"discover"}
Info 33   Project '/dev/null/inferredProject1*' (Inferred) 0
/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/node_modules/typescript/lib/lib.es6.d.ts
/Users/tz/Library/Caches/typescript/node_modules/@types/node/index.d.ts
/Users/tz/Library/Caches/typescript/node_modules/@types/commander/index.d.ts
/Users/tz/Library/Caches/typescript/node_modules/@types/yargs/index.d.ts
/Users/tz/Workspaces/eggjs/common-bin/lib/run.js
/Users/tz/Library/Caches/typescript/node_modules/@types/chalk/index.d.ts
/Users/tz/Workspaces/eggjs/common-bin/lib/program.js
/Users/tz/Library/Caches/typescript/node_modules/@types/debug/index.d.ts
/Users/tz/Workspaces/eggjs/common-bin/lib/helper.js
/Users/tz/Workspaces/eggjs/common-bin/lib/command.js
/Users/tz/Workspaces/eggjs/common-bin/index.js
/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js

-----------------------------------------------
Open files: 
/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js
Info 34   event: {"seq":0,"type":"event","event":"configFileDiag","body":{"triggerFile":"/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js","diagnostics":[]}}
Perf 35   6::open: async elapsed time (in milliseconds) 1780.4235
Info 36   request: {"seq":7,"type":"request","command":"geterr","arguments":{"delay":0,"files":["/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js"]}}
Perf 37   7::geterr: async elapsed time (in milliseconds) 1.7650
Info 38   event: {"seq":0,"type":"event","event":"syntaxDiag","body":{"file":"/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js","diagnostics":[]}}
Info 39   /Users/tz/ti-6176.log is not tsconfig.json
Info 40   /Users/tz/ti-6176.log is not tsconfig.json
Info 41   /Users/tz/ti-6176.log is not tsconfig.json
Info 42   /Users/tz/ti-6176.log is not tsconfig.json
Info 43   /Users/tz/ti-6176.log is not tsconfig.json
Info 44   /Users/tz/ti-6176.log is not tsconfig.json
Info 45   event: {"seq":0,"type":"event","event":"semanticDiag","body":{"file":"/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js","diagnostics":[]}}
Info 46   Sending request: {"projectName":"/dev/null/inferredProject1*","fileNames":["/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/node_modules/typescript/lib/lib.es6.d.ts","/Users/tz/Workspaces/eggjs/common-bin/lib/run.js","/Users/tz/Workspaces/eggjs/common-bin/lib/program.js","/Users/tz/Workspaces/eggjs/common-bin/lib/helper.js","/Users/tz/Workspaces/eggjs/common-bin/lib/command.js","/Users/tz/Workspaces/eggjs/common-bin/index.js","/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js"],"compilerOptions":{"module":1,"target":2,"allowSyntheticDefaultImports":true,"allowNonTsExtensions":true,"allowJs":true,"jsx":1,"noEmitForJsFiles":true},"typeAcquisition":{"enable":true,"include":[],"exclude":[]},"unresolvedImports":["assert","child_process","child_process","crypto","events","fs","http","net","path","path","readline","stream","tls"],"projectRootPath":"","kind":"discover"}
Info 47   /Users/tz/ti-6176.log is not tsconfig.json
Info 48   /Users/tz/.zsh_history.LOCK is not tsconfig.json
Info 49   /Users/tz/.zsh_history.LOCK is not tsconfig.json
Info 50   /Users/tz/.zsh_history.LOCK is not tsconfig.json
```

**Comment by mjbvz** [comment]
Thanks @atian25. There may also be a `ts-xxx.log` file in the same directory as the `tsserver.log` that could help us investigate the typings installer issue. If you see one, please also share it

@mhegazy Do you see anything odd in this log? The two things that stood out to me: 

* `Add watcher for ...` being called up to the root directory. However I see this in my logs as well without seeing the perf issue.
*  `... is not tsconfig.json`. I do not see this in any of my logs and am not sure why it is being triggered for these files.

**Comment by atian25** [comment]
@mjbvz  only `ti-6176.log` and `tsserver.log`

```bash
Global cache location '/Users/tz/Library/Caches/typescript', safe file path '/applications/visual studio code.app/contents/resources/app/extensions/typescript/node_modules/typescript/lib/typingsafelist.json'
Processing cache location '/Users/tz/Library/Caches/typescript'
Trying to find '/Users/tz/Library/Caches/typescript/package.json'...
Loaded content of '/Users/tz/Library/Caches/typescript/package.json': {"devDependencies":{"@types/ansi-styles":"^2.0.30","@types/babel-code-frame":"^6.20.0","@types/babel-core":"^6.7.14","@types/babel-generator":"^6.7.14","@types/babel-template":"^6.7.14","@types/babel-traverse":"^6.7.15","@types/babel-types":"^6.7.14","@types/babylon":"^6.7.15","@types/benchmark":"^1.0.30","@types/camelcase":"0.0.30","@types/chalk":"^0.4.31","@types/colors":"^1.1.1","@types/commander":"^2.3.31","@types/content-type":"^1.0.33","@types/convert-source-map":"^1.3.32","@types/core-js":"^0.9.35","@types/debug":"0.0.29","@types/decamelize":"0.0.30","@types/depd":"^1.1.30","@types/detect-indent":"^0.1.30","@types/echarts":"0.0.5","@types/ejs":"^2.3.33","@types/escape-html":"0.0.19","@types/escape-string-regexp":"0.0.30","@types/extend":"^2.0.30","@types/fs-extra":"0.0.37","@types/glob":"^5.0.30","@types/graceful-fs":"^2.0.29","@types/inherits":"0.0.29","@types/ini":"^1.3.29","@types/inquirer":"0.0.32","@types/invariant":"^2.2.28","@types/is-finite":"^1.0.0","@types/istanbul":"^0.4.29","@types/jsesc":"^0.4.29","@types/json5":"0.0.29","@types/koa":"^2.0.37","@types/load-json-file":"^2.0.5","@types/lodash":"^4.14.52","@types/long":"^3.0.31","@types/merge-descriptors":"0.0.29","@types/mime":"0.0.29","@types/minimatch":"^2.0.29","@types/minimist":"^1.2.0","@types/mkdirp":"^0.3.29","@types/mocha":"^2.2.39","@types/ms":"^0.7.29","@types/mz":"0.0.30","@types/node":"^7.0.4","@types/number-is-nan":"^1.0.0","@types/nunjucks":"0.0.32","@types/once":"^1.3.28","@types/os-homedir":"^1.0.0","@types/os-locale":"^1.4.29","@types/os-tmpdir":"^1.0.0","@types/path-exists":"^1.0.29","@types/path-is-absolute":"^1.0.0","@types/pify":"0.0.28","@types/power-assert":"^1.4.29","@types/qs":"^6.2.30","@types/rimraf":"0.0.28","@types/semver":"^5.3.30","@types/should":"^8.1.30","@types/sinon":"^1.16.35","@types/source-map":"^0.5.0","@types/source-map-support":"^0.2.28","@types/statuses":"^1.2.28","@types/strip-ansi":"^3.0.0","@types/strip-bom":"^3.0.0","@types/superagent":"^2.0.36","@types/supertest":"^2.0.0","@types/supports-color":"^3.1.0","@types/through":"0.0.28","@types/through2":"^2.0.32","@types/vinyl-fs":"0.0.28","@types/wrap-ansi":"^2.0.14","@types/yargs":"^6.6.0"}}
Adding entry into typings cache: 'ansi-styles' => '/Users/tz/Library/Caches/typescript/node_modules/@types/ansi-styles/index.d.ts'
Adding entry into typings cache: 'babel-code-frame' => '/Users/tz/Library/Caches/typescript/node_modules/@types/babel-code-frame/index.d.ts'
Adding entry into typings cache: 'babel-core' => '/Users/tz/Library/Caches/typescript/node_modules/@types/babel-core/index.d.ts'
Adding entry into typings cache: 'babel-generator' => '/Users/tz/Library/Caches/typescript/node_modules/@types/babel-generator/index.d.ts'
Adding entry into typings cache: 'babel-template' => '/Users/tz/Library/Caches/typescript/node_modules/@types/babel-template/index.d.ts'
Adding entry into typings cache: 'babel-traverse' => '/Users/tz/Library/Caches/typescript/node_modules/@types/babel-traverse/index.d.ts'
Adding entry into typings cache: 'babel-types' => '/Users/tz/Library/Caches/typescript/node_modules/@types/babel-types/index.d.ts'
Adding entry into typings cache: 'babylon' => '/Users/tz/Library/Caches/typescript/node_modules/@types/babylon/index.d.ts'
Adding entry into typings cache: 'benchmark' => '/Users/tz/Library/Caches/typescript/node_modules/@types/benchmark/index.d.ts'
Adding entry into typings cache: 'camelcase' => '/Users/tz/Library/Caches/typescript/node_modules/@types/camelcase/index.d.ts'
Adding entry into typings cache: 'chalk' => '/Users/tz/Library/Caches/typescript/node_modules/@types/chalk/index.d.ts'
Adding entry into typings cache: 'colors' => '/Users/tz/Library/Caches/typescript/node_modules/@types/colors/index.d.ts'
Adding entry into typings cache: 'commander' => '/Users/tz/Library/Caches/typescript/node_modules/@types/commander/index.d.ts'
Adding entry into typings cache: 'content-type' => '/Users/tz/Library/Caches/typescript/node_modules/@types/content-type/index.d.ts'
Adding entry into typings cache: 'convert-source-map' => '/Users/tz/Library/Caches/typescript/node_modules/@types/convert-source-map/index.d.ts'
Adding entry into typings cache: 'core-js' => '/Users/tz/Library/Caches/typescript/node_modules/@types/core-js/index.d.ts'
Adding entry into typings cache: 'debug' => '/Users/tz/Library/Caches/typescript/node_modules/@types/debug/index.d.ts'
Adding entry into typings cache: 'decamelize' => '/Users/tz/Library/Caches/typescript/node_modules/@types/decamelize/index.d.ts'
Adding entry into typings cache: 'depd' => '/Users/tz/Library/Caches/typescript/node_modules/@types/depd/index.d.ts'
Adding entry into typings cache: 'detect-indent' => '/Users/tz/Library/Caches/typescript/node_modules/@types/detect-indent/index.d.ts'
Adding entry into typings cache: 'echarts' => '/Users/tz/Library/Caches/typescript/node_modules/@types/echarts/index.d.ts'
Adding entry into typings cache: 'ejs' => '/Users/tz/Library/Caches/typescript/node_modules/@types/ejs/index.d.ts'
Adding entry into typings cache: 'escape-html' => '/Users/tz/Library/Caches/typescript/node_modules/@types/escape-html/index.d.ts'
Adding entry into typings cache: 'escape-string-regexp' => '/Users/tz/Library/Caches/typescript/node_modules/@types/escape-string-regexp/index.d.ts'
Adding entry into typings cache: 'extend' => '/Users/tz/Library/Caches/typescript/node_modules/@types/extend/index.d.ts'
Adding entry into typings cache: 'fs-extra' => '/Users/tz/Library/Caches/typescript/node_modules/@types/fs-extra/index.d.ts'
Adding entry into typings cache: 'glob' => '/Users/tz/Library/Caches/typescript/node_modules/@types/glob/index.d.ts'
Adding entry into typings cache: 'graceful-fs' => '/Users/tz/Library/Caches/typescript/node_modules/@types/graceful-fs/index.d.ts'
Adding entry into typings cache: 'inherits' => '/Users/tz/Library/Caches/typescript/node_modules/@types/inherits/index.d.ts'
Adding entry into typings cache: 'ini' => '/Users/tz/Library/Caches/typescript/node_modules/@types/ini/index.d.ts'
Adding entry into typings cache: 'inquirer' => '/Users/tz/Library/Caches/typescript/node_modules/@types/inquirer/index.d.ts'
Adding entry into typings cache: 'invariant' => '/Users/tz/Library/Caches/typescript/node_modules/@types/invariant/index.d.ts'
Adding entry into typings cache: 'is-finite' => '/Users/tz/Library/Caches/typescript/node_modules/@types/is-finite/index.d.ts'
Adding entry into typings cache: 'istanbul' => '/Users/tz/Library/Caches/typescript/node_modules/@types/istanbul/index.d.ts'
Adding entry into typings cache: 'jsesc' => '/Users/tz/Library/Caches/typescript/node_modules/@types/jsesc/index.d.ts'
Adding entry into typings cache: 'json5' => '/Users/tz/Library/Caches/typescript/node_modules/@types/json5/index.d.ts'
Adding entry into typings cache: 'koa' => '/Users/tz/Library/Caches/typescript/node_modules/@types/koa/index.d.ts'
Adding entry into typings cache: 'load-json-file' => '/Users/tz/Library/Caches/typescript/node_modules/@types/load-json-file/index.d.ts'
Adding entry into typings cache: 'lodash' => '/Users/tz/Library/Caches/typescript/node_modules/@types/lodash/index.d.ts'
Adding entry into typings cache: 'long' => '/Users/tz/Library/Caches/typescript/node_modules/@types/long/index.d.ts'
Adding entry into typings cache: 'merge-descriptors' => '/Users/tz/Library/Caches/typescript/node_modules/@types/merge-descriptors/index.d.ts'
Adding entry into typings cache: 'mime' => '/Users/tz/Library/Caches/typescript/node_modules/@types/mime/index.d.ts'
Adding entry into typings cache: 'minimatch' => '/Users/tz/Library/Caches/typescript/node_modules/@types/minimatch/index.d.ts'
Adding entry into typings cache: 'minimist' => '/Users/tz/Library/Caches/typescript/node_modules/@types/minimist/index.d.ts'
Adding entry into typings cache: 'mkdirp' => '/Users/tz/Library/Caches/typescript/node_modules/@types/mkdirp/index.d.ts'
Adding entry into typings cache: 'mocha' => '/Users/tz/Library/Caches/typescript/node_modules/@types/mocha/index.d.ts'
Adding entry into typings cache: 'ms' => '/Users/tz/Library/Caches/typescript/node_modules/@types/ms/index.d.ts'
Adding entry into typings cache: 'mz' => '/Users/tz/Library/Caches/typescript/node_modules/@types/mz/index.d.ts'
Adding entry into typings cache: 'node' => '/Users/tz/Library/Caches/typescript/node_modules/@types/node/index.d.ts'
Adding entry into typings cache: 'number-is-nan' => '/Users/tz/Library/Caches/typescript/node_modules/@types/number-is-nan/index.d.ts'
Adding entry into typings cache: 'nunjucks' => '/Users/tz/Library/Caches/typescript/node_modules/@types/nunjucks/index.d.ts'
Adding entry into typings cache: 'once' => '/Users/tz/Library/Caches/typescript/node_modules/@types/once/index.d.ts'
Adding entry into typings cache: 'os-homedir' => '/Users/tz/Library/Caches/typescript/node_modules/@types/os-homedir/index.d.ts'
Adding entry into typings cache: 'os-locale' => '/Users/tz/Library/Caches/typescript/node_modules/@types/os-locale/index.d.ts'
Adding entry into typings cache: 'os-tmpdir' => '/Users/tz/Library/Caches/typescript/node_modules/@types/os-tmpdir/index.d.ts'
Adding entry into typings cache: 'path-exists' => '/Users/tz/Library/Caches/typescript/node_modules/@types/path-exists/index.d.ts'
Adding entry into typings cache: 'path-is-absolute' => '/Users/tz/Library/Caches/typescript/node_modules/@types/path-is-absolute/index.d.ts'
Adding entry into typings cache: 'pify' => '/Users/tz/Library/Caches/typescript/node_modules/@types/pify/index.d.ts'
Adding entry into typings cache: 'power-assert' => '/Users/tz/Library/Caches/typescript/node_modules/@types/power-assert/index.d.ts'
Adding entry into typings cache: 'qs' => '/Users/tz/Library/Caches/typescript/node_modules/@types/qs/index.d.ts'
Adding entry into typings cache: 'rimraf' => '/Users/tz/Library/Caches/typescript/node_modules/@types/rimraf/index.d.ts'
Adding entry into typings cache: 'semver' => '/Users/tz/Library/Caches/typescript/node_modules/@types/semver/index.d.ts'
Adding entry into typings cache: 'should' => '/Users/tz/Library/Caches/typescript/node_modules/@types/should/index.d.ts'
Adding entry into typings cache: 'sinon' => '/Users/tz/Library/Caches/typescript/node_modules/@types/sinon/index.d.ts'
Adding entry into typings cache: 'source-map' => '/Users/tz/Library/Caches/typescript/node_modules/@types/source-map/index.d.ts'
Adding entry into typings cache: 'source-map-support' => '/Users/tz/Library/Caches/typescript/node_modules/@types/source-map-support/index.d.ts'
Adding entry into typings cache: 'statuses' => '/Users/tz/Library/Caches/typescript/node_modules/@types/statuses/index.d.ts'
Adding entry into typings cache: 'strip-ansi' => '/Users/tz/Library/Caches/typescript/node_modules/@types/strip-ansi/index.d.ts'
Adding entry into typings cache: 'strip-bom' => '/Users/tz/Library/Caches/typescript/node_modules/@types/strip-bom/index.d.ts'
Adding entry into typings cache: 'superagent' => '/Users/tz/Library/Caches/typescript/node_modules/@types/superagent/index.d.ts'
Adding entry into typings cache: 'supertest' => '/Users/tz/Library/Caches/typescript/node_modules/@types/supertest/index.d.ts'
Adding entry into typings cache: 'supports-color' => '/Users/tz/Library/Caches/typescript/node_modules/@types/supports-color/index.d.ts'
Adding entry into typings cache: 'through' => '/Users/tz/Library/Caches/typescript/node_modules/@types/through/index.d.ts'
Adding entry into typings cache: 'through2' => '/Users/tz/Library/Caches/typescript/node_modules/@types/through2/index.d.ts'
Adding entry into typings cache: 'vinyl-fs' => '/Users/tz/Library/Caches/typescript/node_modules/@types/vinyl-fs/index.d.ts'
Adding entry into typings cache: 'wrap-ansi' => '/Users/tz/Library/Caches/typescript/node_modules/@types/wrap-ansi/index.d.ts'
Adding entry into typings cache: 'yargs' => '/Users/tz/Library/Caches/typescript/node_modules/@types/yargs/index.d.ts'
Finished processing cache location '/Users/tz/Library/Caches/typescript'
Process id: 6183
Npm config file: /Users/tz/Library/Caches/typescript/package.json
Updating types-registry npm package...
Got install request {"projectName":"/dev/null/inferredProject1*","fileNames":["/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/node_modules/typescript/lib/lib.es6.d.ts","/Users/tz/Workspaces/eggjs/common-bin/lib/run.js","/Users/tz/Workspaces/eggjs/common-bin/lib/program.js","/Users/tz/Workspaces/eggjs/common-bin/lib/helper.js","/Users/tz/Workspaces/eggjs/common-bin/lib/command.js","/Users/tz/Workspaces/eggjs/common-bin/index.js","/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js"],"compilerOptions":{"module":1,"target":2,"allowSyntheticDefaultImports":true,"allowNonTsExtensions":true,"allowJs":true,"jsx":1,"noEmitForJsFiles":true},"typeAcquisition":{"enable":true,"include":[],"exclude":[]},"unresolvedImports":["assert","child_process","child_process","crypto","events","fs","http","net","path","path","readline","stream","tls"],"projectRootPath":"","kind":"discover"}
```

**Comment by atian25** [comment]
` is not tsconfig.json` seems to be append when exec `code ~` then select file at tree list

```bash
-----------------------------------------------
Open files: 
/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js
Info 34   event: {"seq":0,"type":"event","event":"configFileDiag","body":{"triggerFile":"/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js","diagnostics":[]}}
Perf 35   6::open: async elapsed time (in milliseconds) 1780.4235
Info 36   request: {"seq":7,"type":"request","command":"geterr","arguments":{"delay":0,"files":["/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js"]}}
Perf 37   7::geterr: async elapsed time (in milliseconds) 1.7650
Info 38   event: {"seq":0,"type":"event","event":"syntaxDiag","body":{"file":"/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js","diagnostics":[]}}
Info 39   /Users/tz/ti-6176.log is not tsconfig.json
Info 40   /Users/tz/ti-6176.log is not tsconfig.json
Info 41   /Users/tz/ti-6176.log is not tsconfig.json
Info 42   /Users/tz/ti-6176.log is not tsconfig.json
Info 43   /Users/tz/ti-6176.log is not tsconfig.json
Info 44   /Users/tz/ti-6176.log is not tsconfig.json
Info 45   event: {"seq":0,"type":"event","event":"semanticDiag","body":{"file":"/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js","diagnostics":[]}}
Info 46   Sending request: {"projectName":"/dev/null/inferredProject1*","fileNames":["/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/node_modules/typescript/lib/lib.es6.d.ts","/Users/tz/Workspaces/eggjs/common-bin/lib/run.js","/Users/tz/Workspaces/eggjs/common-bin/lib/program.js","/Users/tz/Workspaces/eggjs/common-bin/lib/helper.js","/Users/tz/Workspaces/eggjs/common-bin/lib/command.js","/Users/tz/Workspaces/eggjs/common-bin/index.js","/Users/tz/Workspaces/eggjs/common-bin/test/fixtures/my-bin/lib/my_program.js"],"compilerOptions":{"module":1,"target":2,"allowSyntheticDefaultImports":true,"allowNonTsExtensions":true,"allowJs":true,"jsx":1,"noEmitForJsFiles":true},"typeAcquisition":{"enable":true,"include":[],"exclude":[]},"unresolvedImports":["assert","child_process","child_process","crypto","events","fs","http","net","path","path","readline","stream","tls"],"projectRootPath":"","kind":"discover"}
Info 47   /Users/tz/ti-6176.log is not tsconfig.json
Info 48   /Users/tz/.zsh_history.LOCK is not tsconfig.json
Info 49   /Users/tz/.zsh_history.LOCK is not tsconfig.json
Info 50   /Users/tz/.zsh_history.LOCK is not tsconfig.json
Info 51   /Users/tz/.zsh_history.LOCK is not tsconfig.json
Info 52   /Users/tz/.zsh_history.LOCK is not tsconfig.json
Info 53   /Users/tz/.zsh_history.LOCK is not tsconfig.json
Info 54   /Users/tz/.zsh_history.LOCK is not tsconfig.json
Info 55   /Users/tz/.zsh_history.LOCK is not tsconfig.json
```

**Comment by atian25** [comment]
![image](https://cloud.githubusercontent.com/assets/227713/23387223/3e589da4-fd96-11e6-8d6c-6903bd83b414.png)

another problem, seems to trigger by my settings `search.exclude`

```
➜  ~ ps -ef|grep find
    0   201     1   0 Thu01PM ??         0:01.23 /usr/libexec/findmydeviced
  501  7729     1   0  9:08AM ??         2:03.39 find -L . -not ( ( -name node_modules -o -name .git -o -name .svn -o -name .hg -o -name .DS_Store -o -name .idea -o -path ./.vscode-test ) -prune ) -type f
  501  7730     1   0  9:08AM ??         2:02.49 find -L . -not ( ( -name node_modules -o -name .git -o -name .svn -o -name .hg -o -name .DS_Store -o -name .idea -o -path ./.vscode-test ) -prune ) -type f
  501  7731     1   0  9:08AM ??         2:02.36 find -L . -not ( ( -name node_modules -o -name .git -o -name .svn -o -name .hg -o -name .DS_Store -o -name .idea -o -path ./.vscode-test ) -prune ) -type f
```

**Comment by atian25** [comment]
@mjbvz do you still need more info ?

**Comment by djut** [comment]
Hot in my domain and you are abusing it

Sent from my iPhone

On Mar 1, 2017, at 5:34 PM, TZ | 天猪 <notifications@github.com<mailto:notifications@github.com>> wrote:


@mjbvz<https://github.com/mjbvz> do you still need more info ?

**Comment by mjbvz** [comment]
Adding @chrmarti for the `find` command part

@mhegazy, let's look through the logs tomorrow and see if anything stands out that could be causing this

**Comment by whxaxes** [comment]
workaround: `ps -ef|grep typingsInstall | awk '{print $2}' | xargs kill`

**Comment by mjbvz** [comment]
Thank you @atian25 for providing all this information. I've opened https://github.com/Microsoft/TypeScript/issues/14655 against TypeScript to track the TSServer/TypingsInstaller part of this bug

**Comment by mhegazy** [comment]
I think the issue is we are watching folders like `/Users`, every time there is a change the server is notified. the message `/Users/tz/ti-6176.log is not tsconfig.json` indicates that the server is processing a change notification on this folder; in this specific case, seems like the log for the typingsInstaller.

@whxaxes does adding a jsconfig.json to the root of your project alleviate the CPU utilization?

**Comment by syusukexrj** [comment]
@atian25  another workaround, rename typingsInstaller.js.

**Comment by medikoo** [comment]
Same here, and I don't even work with TS files.
If it's a side effect of a feature related purely to handling of TS files, then it's surprising it affects editor as a whole.

**Comment by chrmarti** [comment]
Reassigning. @atian25 Could you open a separate issue for `find` processes if that is still an issue?

**Comment by atian25** [comment]
@chrmarti  only trigger twice, seems to be trigger by `typingsInstall`, but these time I kill `typingsInstall` when cpu rush, so maybe it can't trigger again.

**Comment by mjbvz** [comment]
I've opened https://github.com/Microsoft/TypeScript/issues/15301 to track the hang in the Typings installer. It is not clear what is causing this hang but I have been able to reproduce it a few times

VSCode 1.12 with TypeScript 2.3 also fixes a few problems that could result in similar issues. Please try out the latest insiders builds to see if these resolve your issue

If you continue hitting the hang in typings installer, just set `"typescript.disableAutomaticTypeAcquisition": true` to disable this feature.

**Comment by atian25** [comment]
Still got trouble at 1.12.2 without  `disableAutomaticTypeAcquisition`

**Comment by mhegazy** [comment]
@atian25 does this happen working on any .js file or a specific project of yours?

**Comment by atian25** [comment]
@mhegazy not specific.

some addition info:

all these project is nodejs project, and `node_modules` is not install by `npm` but `npminstall`(https://github.com/cnpm/npminstall) which sometime will cause link problem.

**Comment by mrmengjun** [comment]
add these in your vscode settings ->
 "typescript.disableAutomaticTypeAcquisition": true
